sailor/pai/alert.py

```diff
@@ -0,0 +1,133 @@
+            'model_id': ('ModelID', None, None, None),
+            'model_name': ('ModelName', None, None, None),
+            'notification_id': ('NotificationId', None, None, None),
+            'operator_id': ('OperatorID', None, None, None),
+            'operator_name': ('OperatorName', None, None, None),
+            'planner_group': ('PlannerGroup', None, None, None),
+            'planning_plant': ('PlanningPlant', None, None, None),
+            'processor': ('Processor', None, None, None),
+            'ref_alert_type_id': ('RefAlertTypeId', None, None, None),
+            'serial_number': ('SerialNumber', None, None, None),
+            'severity_code': ('SeverityCode', None, None, None),
+            'source': ('Source', None, None, None),
+            'template_id': ('TemplateID', None, None, None),
+            'status_code': ('StatusCode', None, None, None),
+            'template_name': ('TemplateName', None, None, None),
+            'top_equipment_description': ('TopEquipmentDescription', None, None, None),
+            'top_equipment_id': ('TopEquipmentID', None, None, None),
+            'top_equipment_name': ('TopEquipmentName', None, None, None),
+            'top_functional_location_id': ('TopFunctionalLocationID', None, None, None),
+            'top_functional_location_name': ('TopFunctionalLocationName', None, None, None),
+            'top_functional_location_description': ('TopFunctionalLocationDescription', None, None, None),
+            'triggered_on': ('TriggeredOn', _odata_to_timestamp_parser('TriggeredOn', unit='s'), None, None)
+        }
+
+
+class AlertSet(ResultSet):
+    """Class representing a group of Alerts."""
+
+    _element_type = Alert
+    _method_defaults = {
+        'plot_distribution': {
+            'by': 'type',
+        },
+    }
+
+
+def find_alerts(*, extended_filters=(), **kwargs) -> AlertSet:
+    """
+    Fetch Alerts from PredictiveAssetInsights (PAI) with the applied filters, return an AlertSet.
+
+    This method supports the common filter language explained at :ref:`filter`.
+
+    Parameters
+    ----------
+    extended_filters
+        See :ref:`filter`.
+    **kwargs
+        See :ref:`filter`.
+
+    Examples
+    --------
+    Get all Alerts with the type 'MyAlertType'::
+
+        find_alerts(type='MyAlertType')
+
+    Get all Error(severity code=10) and Information(severity code=1) alerts::
+
+        find_equipment(severity_code=[10, 1])
+    """
+    unbreakable_filters, breakable_filters = \
+        _parse_filter_parameters(kwargs, extended_filters, Alert._get_legacy_mapping())
+
+    endpoint_url = _pai_application_url() + ALERTS_READ_PATH
+    objects = []
+    object_list = _fetch_data(endpoint_url, unbreakable_filters, breakable_filters, 'predictive_asset_insights')
+    for odata_result in object_list:
+        for element in odata_result['d']['results']:
+            objects.append(element)
+    return AlertSet([Alert(obj) for obj in objects],
```

Comment:
The return value of `_fetch_data` should only contain a single result (so in general no looping should be necessary).

However, this is only the case for the Alerts. The results of the Assetcentral APIs are lists of objects and not a single object with the `d`. But that's okay, depending which OData Spec version the API adheres to (things have changed). 
We could improve the `_fetch_data` function to automatically "find" the result list in the response and return it. This way we could catch errors early and
Would that make sense at all? Do it here or add it to the refactoring list? @atrejut

Reply:
makes sense, let's add it to the refactoring list though.